test/HatraceSpec.hs

```diff
@@ -660,6 +660,32 @@ spec = before_ assertNoChildren $ do
          -- in that case, this test needs to use a manual implementation.
          syscalls `shouldSatisfy` (\xs -> KnownSyscall Syscall_dup2 `elem` xs)
 
+       it "sees the correct dup3 arguments" $ do
+         let dup3Syscall = "example-programs-build/dup3"
+         callProcess "make" ["--quiet", dup3Syscall]
+         argv <- procToArgv dup3Syscall []
+         (exitCode, events) <-
+           sourceTraceForkExecvFullPathWithSink argv $
+             syscallExitDetailsOnlyConduit .| CL.consume
```

Comment:
Ideally I would have used a bogus dup3 syscall in the test program, like `dup3(0, 1, 42)`, but it seems like that expression was filtering it as `(2095,Left (KnownSyscall Syscall_dup3,ERRNO 22))`. Unfortunately I'm not familiar enough with conduit to figure out how to let it keep the EnterDetails for that erred syscall.